proxy/proxy/client.go

```diff
@@ -166,6 +170,11 @@ func (c *Client) refreshCfg(instance string) (addr string, cfg *tls.Config, vers
 		throttle = DefaultRefreshCfgThrottle
 	}
 
+	refreshCertBuffer := c.RefreshCertBuffer
```

Comment:
Are we assuming that `refreshCfgL` is protecting this value as well?

Reply:
Yes... TBH I didn't even notice that. But yes I was thinking this would be treated the exact same as `c.RefreshCfgThrottle`.

I'm OK to not add this as a configurable value at all. Changing the default to 5 mins as you suggested is probably plenty of time, it just seemed weird to be able to configure `RefreshCfgThrottle` without being able to configure this value.

Reply:
Seems fine to include it, I just have one nit that'd I'd prefer it be `RefreshCfgBuffer` instead of `RefreshCertBuffer` - it's a little obvious then that `refreshCfg` protects `RefreshCfgThrottle` and `RefreshCfgBuffer`. 

If possible, it would also be useful to move these two variables under the lock as well, which helps signal this as well.

Reply:
Made those changes, thanks for suggesting them.